packages/story-editor/src/components/fontCheck/fontCheckDialog.js

```diff
@@ -0,0 +1,191 @@
+import { TranslateWithMarkup, __ } from '@googleforcreators/i18n';
+import { trackClick } from '@googleforcreators/tracking';
+import styled from 'styled-components';
+import { useState, useCallback, useMemo } from '@googleforcreators/react';
+import {
+  Button,
+  BUTTON_SIZES,
+  BUTTON_TYPES,
+  Text,
+  THEME_CONSTANTS,
+  Icons,
+  Link,
+} from '@googleforcreators/design-system';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../app';
+import Dialog from '../dialog';
+import FontPicker from '../panels/design/textStyle/fontPicker';
+import { useConfig } from '../../app/config';
+import { TEXT_ELEMENT_DEFAULT_FONT } from '../../app/font/defaultFonts';
+
+const DialogContent = styled.div`
+  display: grid;
+  row-gap: 21px;
+`;
+
+const ReversedIcon = styled(Icons.ChevronRightSmall)`
+  transform: rotate(180deg);
+`;
+
+const DashboardLink = styled.span`
+  display: flex;
+  color: black;
+  align-self: flex-start;
+  align-items: center;
+  font-weight: normal;
+  margin-right: 40px;
+`;
+
+export const FontCheckDialog = ({ isOpen, onClose, missingFont }) => {
+  const { dashboardLink, dashboardSettingsLink } = useConfig();
+  const defaultFont = useMemo(() => {
+    return {
+      font: {
+        id: TEXT_ELEMENT_DEFAULT_FONT?.family,
+        ...TEXT_ELEMENT_DEFAULT_FONT,
+      },
+    };
+  }, []);
+  const [suggestedFont, setSuggestedFont] = useState(defaultFont);
+  const { updateElementsByFontFamily } = useStory((state) => ({
+    updateElementsByFontFamily: state.actions.updateElementsByFontFamily,
+  }));
+
+  /**
+   * Handles updating suggested font which is passed to the FontPicker
+   *
+   * @param {Object} [evt] Font object.
+   * @return void
+   */
+  const onPushUpdate = useCallback((evt) => {
+    if (evt?.font) {
+      setSuggestedFont({
+        font: { id: evt.font.family, family: evt.font.family },
+      });
+    }
+  }, []);
+
+  const onSettingsClick = useCallback((evt) => {
+    trackClick(evt, 'click_video_optimization_settings');
+  }, []);
+
+  const updateMissingFontWithDefault = useCallback(() => {
+    const { id, ...font } = defaultFont.font;
+    updateElementsByFontFamily({
+      family: missingFont,
+      properties: { font },
+    });
+    onClose();
+  }, [defaultFont, onClose, missingFont, updateElementsByFontFamily]);
+
+  const updateMissingFontWithSelected = useCallback(() => {
+    const { id, ...font } = suggestedFont.font;
+    updateElementsByFontFamily({
+      family: missingFont,
+      properties: { font },
+    });
+    onClose();
+  }, [suggestedFont, onClose, missingFont, updateElementsByFontFamily]);
+
+  return (
+    <Dialog
+      isOpen={isOpen}
+      onClose={onClose}
+      title={__('Missing Fonts', 'web-stories')}
+      contentLabel={__('Missing Fonts', 'web-stories')}
+      actions={
+        <>
+          <Link href={dashboardLink}>
+            <DashboardLink>
+              <ReversedIcon width="32" height="32" />
```

Comment:
Note: Numbers should be passed as such in JSX.

```jsx
              <ReversedIcon width={32} height={32} />
```

But even better, define the width and height in the styled component itself.

```jsx
              <ChevronLeft />
```

And even better, make it work in RTL layout to prevent this:

![Screenshot 2022-03-09 at 10 22 21](https://user-images.githubusercontent.com/841956/157411743-c1edb999-97fd-4f77-a605-a4174b5ee356.png)

Further up in the component, do:

```jsx
const { isRTL } = useConfig();

// ...

const Chevron = isRTL ? ChevronRight : ChevronLeft;
```

And then here:

```suggestion
              <Chevron />
```